File: 백준/Silver/1629. 곱셈/곱셈.py
A, B, C = map(int, input().split())

def find(b):
    if b % 2 == 0:
        a = find(b//2) % C
        res = a * a
        return res%C
    elif b == 1 :
        return A%C
    else:
        a = find(b//2)%C
        res = a * ((a * A)%C)
        return res%C
    

print(find(B))

File: 백준/Silver/1629. 곱셈/곱셈.js
const fs = require("fs");

const input = fs.readFileSync(0, "utf8").trim().split("\n");
const [A, B, C] = input[0].split(" ").map(BigInt)

let move = []
let target = B;

while (target != 1) {
    move.push(target);
    target = target/2n;
}

move = move.sort((a, b) => {return Number(a-b);});

let once = A%C;
let cur = once;

for (let i of move) {
    let res = (cur * cur) % C;

    if (i % 2n == 1) {
        res *= once;
        res %= C;
    }

    cur = res;
    // console.log(i, cur);    
}

console.log(Number(cur));